lib/authz/permissions.go

```diff
@@ -1408,6 +1420,39 @@ func AuthorizeContextWithVerbs(ctx context.Context, log logrus.FieldLogger, auth
 	return authCtx, nil
 }
 
+// CheckAccessToKind will ensure that the user has access to the given verbs for the given kind.
+func (c *Context) CheckAccessToKind(quiet bool, kind string, verbs ...string) error {
```

Comment:
I've always kinda hated seeing random true/false in calls to these and not immediately knowing what it does - while we're changing these signatures can we get rid of this? Could use a custom type with named constants or just export `CheckAccessToKind` and `CheckAccessToKindVerbose` (assuming the verbose option is less common)

Reply:
The quiet field is passed through to `RoleSet.checkAccess` to determine whether we do `Infof("Access to {verb} {resource} in {namespace} denied to {role}: deny rule matched."`. @rosstimothy Are you saying we should just always log to Debug and remove the silent flag? If so, I agree.

Reply:
Opened a follow up PR https://github.com/gravitational/teleport/pull/37771